.teamcity/_self/lib/e2e/Steps.kt

```diff
@@ -0,0 +1,179 @@
+ */
+fun BuildSteps.prepareEnvironment(): ScriptBuildStep {
+	return script {
+		name = "Prepare environment"
+		scriptContent = """
+			#!/bin/bash
+			# Update node
+			. "${'$'}NVM_DIR/nvm.sh" --no-use
+			nvm install
+			set -o errexit
+			set -o nounset
+			set -o pipefail
+
+			export NODE_ENV="test"
+			export PLAYWRIGHT_BROWSERS_PATH=0
+
+			# Install deps
+			yarn workspaces focus wp-e2e-tests @automattic/calypso-e2e
+
+			# Build packages
+			yarn workspace @automattic/calypso-e2e build
+		""".trimIndent()
+		dockerImagePlatform = ScriptBuildStep.ImagePlatform.Linux
+		dockerPull = true
+		dockerImage = "%docker_image_e2e%"
+		dockerRunParameters = "-u %env.UID%"
+	}
+}
+
+/**
+ * Collects the resulting artifacts from a test run.
+ *
+ * This function will collect the following:
+ * 	- Screenshots
+ *	- Logs
+ *	- Trace files
+ */
+fun BuildSteps.collectResults(): ScriptBuildStep {
+	return script {
+		name = "Collect results"
+		executionMode = BuildStep.ExecutionMode.RUN_ON_FAILURE
+		scriptContent = """
+			#!/bin/bash
+			# Update node
+			. "${'$'}NVM_DIR/nvm.sh" --no-use
+			nvm install
+			set -o errexit
+			set -o nounset
+			set -o pipefail
+			set -x
+
+			mkdir -p screenshots
+			find test/e2e -type f -path '*/screenshots/*' -print0 | xargs -r -0 mv -t screenshots
+
+			mkdir -p logs
+			find test/e2e -name '*.log' -print0 | xargs -r -0 tar cvfz logs.tgz
+
+			mkdir -p trace
+			find test/e2e/results -name '*.zip' -print0 | xargs -r -0 mv -t trace
+		""".trimIndent()
+		dockerImagePlatform = ScriptBuildStep.ImagePlatform.Linux
+		dockerPull = true
+		dockerImage = "%docker_image_e2e%"
+		dockerRunParameters = "-u %env.UID%"
+	}
+}
+
+/**
+ *
+ */
+fun BuildSteps.runTests(
+	stepName: String = "",
+	dockerBuildNumber: String = "",
+	testGroup: String = "",
+	envVars: Map<String, String>
+): ScriptBuildStep {
+	return script {
+		name = stepName
+
+		val scriptContentBuilder = StringBuilder()
+
+		scriptContentBuilder.appendLine(
+			"""
+			#!/bin/bash
+			# Update node
+			. "${'$'}NVM_DIR/nvm.sh" --no-use
+			nvm install
+			set -o errexit
+			set -o nounset
+			set -o pipefail
+			set -x
+
+			# Configure bash shell.
+			shopt -s globstar
+
+			""".trimIndent()
+		)
+
+		if (!dockerBuildNumber.isBlank()) {
+			scriptContentBuilder.appendLine(
+				"""
+				chmod +x ./bin/get-calypso-live-url.sh
+				URL=${'$'}(./bin/get-calypso-live-url.sh $dockerBuildNumber)
+				if [[ ${'$'}? -ne 0 ]]; then
+					// Command failed. URL contains stderr
+					echo ${'$'}URL
+					exit 1
+				fi
+
+				""".trimIndent()
+			)
+		}
+
+		scriptContentBuilder.appendLine(
+			"""
+			cd test/e2e
+			mkdir temp
+
+			"""
+		)
+
+		scriptContentBuilder.appendLine(
+			"""
+			export NODE_CONFIG_ENV=test
+			export PLAYWRIGHT_BROWSERS_PATH=0
+			export TEAMCITY_VERSION=2021
+			export DEBUG=pw:api
+			export HEADLESS=false
+			export LOCALE=en
```

Comment:
This var doesn't seem specifically related to TeamCity. Maybe this should be the default value so we don't need to specify it.

Reply:
Changed to set these values as default.